plugin/src/main/java/org/opensearch/ml/model/MLModelAutoReLoader.java

```diff
@@ -0,0 +1,273 @@
+        clusterService
+            .getClusterSettings()
+            .addSettingsUpdateConsumer(ML_COMMONS_MODEL_AUTO_RELOAD_ENABLE, it -> enableAutoReLoadModel = it);
+    }
+
+    /**
+     * the main method: model auto reloading
+     */
+    public void autoReLoadModel() {
+        log.debug("enableAutoReLoadModel {} ", enableAutoReLoadModel);
+
+        // if we don't need to reload automatically, just return without doing anything
+        if (!enableAutoReLoadModel) {
+            return;
+        }
+
+        // At opensearch startup, get local node id, if not ml node,we ignored, just return without doing anything
+        String localNodeId = clusterService.localNode().getId();
+        if (!MLNodeUtils.isMLNode(nodeHelper.getNode(localNodeId))) {
+            return;
+        }
+
+        // According to the node id to get retry times, if more than the maxi retry times, don't need to retry
+        // that the number of unsuccessful reload has reached the maximum number of times, do not need to reload
+        int reTryTimes = getReTryTimes(localNodeId);
+        if (reTryTimes > ML_MODEL_RELOAD_MAX_RETRY_TIMES) {
+            log.debug("have exceeded max retry times, always failure");
+        }
+
+        // auto reload all models of this local node, if it fails, reTryTimes+1, if it succeeds, reTryTimes is cleared to 0
+        CompletableFuture.runAsync(() -> autoReLoadModelByNodeId(localNodeId));
+    }
+
+    /**
+     * auto reload all the models under the node id
+     * the node must be a ml node
+     * @param nodeId node id
+     */
+    @VisibleForTesting
+    void autoReLoadModelByNodeId(String nodeId) {
+        if (!isExistedIndex(ML_TASK_INDEX)) {
+            return;
+        }
+
+        SearchResponse response = queryTask(nodeId);
+
+        SearchHit[] hits = response.getHits().getHits();
+        if (CollectionUtils.isEmpty(hits)) {
+            return;
+        }
+
+        if (hits.length != 1) {
+            log.error("Can't get the latest successful deployment in node id {}", nodeId);
+        }
+
+        int reTryTimes = 0;
+        try (XContentParser parser = createXContentParserFromRegistry(xContentRegistry, hits[0].getSourceRef())) {
+            ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+            MLTask mlTask = MLTask.parse(parser);
+
+            autoReLoadModelByNodeAndModelId(nodeId, mlTask.getModelId());
+        } catch (RuntimeException | IOException e) {
+            reTryTimes++;
+            // Store the latest value of the reTryTimes and node id under the index ".plugins-ml-model-reload"
+            saveLatestReTryTimes(nodeId, reTryTimes);
+            log.error("Can't auto reload model in node id {} ,has try {} times\nThe reason is:{}", nodeId, reTryTimes, e.getMessage());
+        }
+
+        // Store the latest value of the reTryTimes and node id under the index ".plugins-ml-model-reload"
+        saveLatestReTryTimes(nodeId, reTryTimes);
```

Comment:
What if the reload successfully? Do we still need to save retry times or clear it up.

Reply:
in this situation, we should clear `retrytimes` to zero when we reload successfully,so we use  `saveLatestReTryTimes(nodeId, reTryTimes);`to save 0 into it

Reply:
This method will be invoked twice if entered the catch block, remove the invocation in catch block.

Reply:
got it